tools/crates/build_crate.gni

```diff
@@ -3,60 +3,85 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at https://mozilla.org/MPL/2.0/.
 
+import("//brave/build/host_os_constants.gni")
+
 template("build_crate") {
-  crate = target_name
-  assert(
-      defined(invoker["${crate}_exe"]),
-      "Please add ${crate}_exe to //brave/tools/crates/config.gni (or to $crate itself)!")
-  crate_exe = invoker["${crate}_exe"]
-  crate_target_dir = get_path_info(get_path_info(crate_exe, "dir"), "dir")
-
-  action("build_$crate") {
+  if (defined(invoker["crates"])) {
+    crates = invoker["crates"]
+  } else {
+    crates = [ target_name ]
+  }
+
+  if (defined(invoker["executable_outputs"])) {
+    executable_outputs = invoker["executable_outputs"]
+  } else {
+    executable_outputs = [ target_name ]
+  }
+
```

Comment:
```gn
assert(len(crates) == len(executable_outputs))
```

Reply:
A package can potentially generate more than one executable or less than one, so I decided to not add such assert (I thought about this also).